src/ephemeris/shed_tools.py

```diff
@@ -627,10 +637,16 @@ def main():
             parallel_tests=args.parallel_tests,
             test_all_versions=args.test_all_versions,
             client_test_config_path=args.client_test_config,
+            cleanup_histories=args.cleanup_histories,
+            fail_on_error=args.fail_on_error,
         )
     else:
         raise NotImplementedError("This point in the code should not be reached. Please contact the developers.")
 
+    # Validate if all tools installed correctly
+    if args.fail_on_error and install_results and install_results.errored_repositories > 0:
+        log.info("The %s of a tool has failed. Exiting." % args.action)
```

Comment:
log.error would be more appropriate.
Alternatively: since log + exit is usted. It may be better to use `raise`  an error instead.